tests/api_test.py

```diff
@@ -3120,6 +3120,16 @@ def f(_):
     expected = jnp.arange(1) + 1
     self.assertAllClose(ans, expected)
 
+  @parameterized.named_parameters([
+      {"testcase_name": f"{dtype.__name__}", "dtype": dtype}
+      for dtype in jtu.dtypes.all])
```

Comment:
The issue you have in the internal CI is that you cannot determine which backend we are using (via `jtu.dtypes`) until after absl initialization has run. This parameter gets instantiated too early.

This is why we usually use the idiom of skipping unsupported dtypes.

Reply:
If that's the case, then why do these not cause the same issue? https://github.com/google/jax/blob/f80887e69cdac905a7c748622fd6bcd75bbc0cf9/tests/lax_numpy_test.py#L67-L75

Reply:
They do. But I suspect anyone calling that test is setting `--jax_test_dut`:
https://github.com/google/jax/blob/f80887e69cdac905a7c748622fd6bcd75bbc0cf9/jax/_src/test_util.py#L434

Reply:
Can we require that for tests? (it's not our own internal tests that are failing, but someone else using our targets)

Reply:
Yes perhaps. Probably we'd do well to make the logic unconditionally get its "target device" from exactly one source: either flags or "what's registered" but not both. Not necessarily for this PR.